source/common/grpc/google_async_client_impl.cc

```diff
@@ -167,7 +167,24 @@ GoogleAsyncStreamImpl::GoogleAsyncStreamImpl(GoogleAsyncClientImpl& parent,
       service_full_name_(service_full_name), method_name_(method_name), callbacks_(callbacks),
       options_(options), unused_stream_info_(Http::Protocol::Http2, dispatcher_.timeSource(),
                                              Network::ConnectionInfoProviderSharedPtr{},
-                                             StreamInfo::FilterState::LifeSpan::FilterChain) {}
+                                             StreamInfo::FilterState::LifeSpan::FilterChain) {
+  if (options.parent_span_ != nullptr) {
+    const std::string child_span_name =
+        options.child_span_name_.empty()
+            ? absl::StrCat("async ", service_full_name, ".", method_name, " egress")
+            : options.child_span_name_;
+    current_span_ = options.parent_span_->spawnChild(Tracing::EgressConfig::get(), child_span_name,
+                                                     parent.timeSource().systemTime());
+    current_span_->setTag(Tracing::Tags::get().UpstreamAddress, parent.target_uri_);
```

Comment:
[GoogleAsyncRequestImpl](https://github.com/envoyproxy/envoy/blob/99bb5659f8b204068cdb307a898d617d0ee4b5e8/source/common/grpc/google_async_client_impl.cc#L450) is also setting  UpstreamCluster  ----- `current_span_->setTag(Tracing::Tags::get().UpstreamCluster, parent.stat_prefix_)`

I am not able to find out why `UpstreamCluster` was originally set for google gRPC as it doesn't make use of  Envoy’s  upstream connection management like envoy-gRPC.   + @wbpcode / @htuch in case they know its use case.

We probably should have consistent behavior here: EITHER add UpstreamCluster if it does have value OR remove it in googleAsyncRequest

Reply:
Can we factor this to a common library for span annotations so we keep the two gRPC implementations in sync?

Reply:
@cainelli if you could add a TODO for refactor and follow this up, I am personally fine with this PR to unblock you unless @htuch has strong opinion on this.
 
For common library, we can start with something simple like utility functions/class etc.  The key is singe source-of-truth  (when applicable) for both Envoy-gRPC and google-gRPC

Reply:
Thanks @tyxia, I added a comment and fixed the test failing.

Reply:
OK, this is fine, it's not critical, so a TODO works.